packages/web/src/js/FileUploader.ts

```diff
@@ -244,12 +248,20 @@ class FileUploader extends BaseComponent {
         if (item.kind === 'file') {
           const file = item.getAsFile();
           if (file) {
-            this.addToQueue(file);
+            if (count > 0) {
+              this.addToQueue(file);
+              count -= 1;
+            }
           }
         }
       });
     } else {
-      transferFiles.forEach((file) => this.addToQueue(file));
+      transferFiles.forEach((file) => {
```

Comment:
When we are using only `items`, it is necessary to use also files?

> In the JavaScript API DataTransfer, the "files" and "items" properties are two different ways of accessing the data being transferred.
> 
> The "files" property contains a list of File objects representing the files being transferred, if any. This property is typically used in drag and drop interactions where files are being dragged from the user's file system and dropped onto a web page.
> 
> On the other hand, the "items" property contains a list of DataTransferItem objects representing the data being transferred, which can include files, text, or other data types. This property is typically used in copy/paste or cut/paste interactions where the data is coming from a source other than the user's file system.
> 
> In summary, "files" is used for accessing file data specifically, while "items" can be used to access any type of data being transferred, including files.

Reply:
In short: `transferItems` is from all browsers except Safari, and `transferFiles` is from Safari

Reply:
I have checked this in Safari, Chrome, and Firefox and in all cases, there is used `transferItems`.

Anyway `if (Array.from(undefined || [])) { console.log('rimmer') } else { console.log('kryton') } // output: rimmer`.
The `[]` is still `true` in every case of using `undefined`, `null`, or `''`.

So I am not convinced that the `else` is reachable and thus working or needed.

Reply:
<img width="819" alt="Screenshot 2023-05-09 at 22 13 57" src="https://github.com/lmc-eu/spirit-design-system/assets/2481010/13054730-8169-4209-8fc3-9ebf5ee9105a">
<img width="828" alt="Screenshot 2023-05-09 at 22 14 29" src="https://github.com/lmc-eu/spirit-design-system/assets/2481010/eb79f617-8acc-42d3-9fab-6577e37ad37b">
<img width="957" alt="Screenshot 2023-05-09 at 22 14 53" src="https://github.com/lmc-eu/spirit-design-system/assets/2481010/7efd426e-09d8-4f27-a315-dd6a394d1361">

Reply:
Look, I really don't care if it's here or not. I just wanted to cover all the cases where it might come up here. If you don't like it, I'll delete it, but I really don't want to deal with the case in the future if it doesn't end up here. So, despite my objections, I'll just delete it, ok?

Reply:
I better set the condition to `if (event?.dataTransfer?.items)`